openpype/lib/delivery.py

```diff
@@ -13,18 +13,30 @@ def collect_frames(files):
         Uses clique as most precise solution
 
         Args:
-            files(list): list of source paths
+            files(list) or (set with single value): list of source paths
         Returns:
             (dict): {'/asset/subset_v001.0001.png': '0001', ....}
     """
     collections, remainder = clique.assemble(files, minimum_items=1)
 
+    real_file_name = None
+    if len(files) == 1:
+        real_file_name = list(files)[0]
+
     sources_and_frames = {}
     if collections:
         for collection in collections:
             src_head = collection.head
             src_tail = collection.tail
 
+            if src_head.endswith("_v"):
```

Comment:
This seems rather hardcoded. Anatomy does allow productions to override the filename structure so that the version string could be elsewhere in the filename?

It could be `v001_asset01_subset.exr`? - The number picked up would still not be the frame number?

Reply:
All your created tests show the frame numbers are separated with a `.` like `file.0001.exr`. If we ALWAYS know that's the case, we can provide a more explicit pattern to `clique.assemble` which would be amazing.

But we'd need to ensure we never produce content like `file_0001.exr`.

Reply:
You are right, I tried to create more generic, though safe, pattern. Updated tests.
I think that this function is used (or at least I am using it) in case that I don't know anything about format of files (in my case files are produced locally by AE render). 
I am still depending that version substring starts with 'v', following with any number of numeric characters.